### DISCUSSION TITLE
Pre-Shared key

### DISCUSSION THREAD

**Discussion by sigma2017** [wg-easy/wg-easy#819] [Q&A]
Hello,
Is it possible to make wg-easy to work without Pre-Shared key Maybe an additional parameter into docker run?
I read that the pre-shared key is not necessary, wireguard is secure enough without.
I've noticed from manual installation that the Pre-Shared key is slowing down the speed.
I have removed the Pre-Shared from wg0.conf and wg0.json but the container will fail at startup:

```
$ wg-quick down wg0
$ wg-quick up wg0
Error: Command failed: wg-quick up wg0
[#]
[#] ip link add wg0 type wireguard
[#] wg setconf wg0 /dev/fd/63
Key is not the correct length or format: `undefined'
Configuration parsing error
[#] ip link delete dev wg0

    at ChildProcess.exithandler (node:child_process:422:12)
    at ChildProcess.emit (node:events:517:28)
    at maybeClose (node:internal/child_process:1098:16)
    at ChildProcess._handle.onexit (node:internal/child_process:303:5) {
  code: 1,
  killed: false,
  signal: null,
  cmd: 'wg-quick up wg0'
```
I'm sure than the speed loose is caused by the pre-shared key, not because wg-easy runs with docker....
Thank you!

**Comment by peterlewis** [comment]
No plans to develop this feature. From what I can see (and as we know, Reddit is always right... 🙃) using PSK should not be of detriment to performance:

https://www.reddit.com/r/WireGuard/comments/agys0s/comment/hselmlt/

Either way though, no plans, currently. Thanks!

**Comment by sigma2017** [comment]
I don't know where the performance drops comes from, either from PSK (maybe not) or from the fact is running under docker layer, the thing is its a bit slower compared with native install. I would say around 20% performance loss.
wg-easy is simply outstanding, either way, congrats for your work!
Thanks for the reply.
Best regards.

**Comment by sigma2017** [comment]
Hello,
I made some additional tests and indeed PSK doesn't have anything to do with the performance.
The performance drop comes from the fact that wg-easy runs inside a docker container.
You were right, thanks again.

**Comment by FadeFx** [comment]
I think it is possible to run wg-easy in hostmode, that means it has only the interface I docker and actually configures a wireguard interface on the host directly. Iirc this requires to give wg-easy container access to /etc/wireguard/ thou...
